addon/services/options.js

```diff
@@ -13,17 +13,17 @@ export default Service.extend({
 
   init() {
     this._super(...arguments);
-    this._data = new EmberObject();
 
-    this._data.set("overrides", [
+    this._data = new EmberObject();
+    this._overrides = [
```

Comment:
Wouldn't it be much simpler to use an object indexed by component name instead of a list? Then you can replace all the finding and splicing by simple assignments and member checks.